app/src/main/java/io/appium/uiautomator2/model/settings/SnapshotMaxDepth.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.model.settings;
+
+public class SnapshotMaxDepth extends AbstractSetting<Integer> {
+    public static final String SETTING_NAME = "snapshotMaxDepth";
+    // Set DEFAULT_VALUE as 70 to avoid StackOverflow from infinite recursion
+    // https://github.com/appium/appium/issues/12545
+    // https://github.com/appium/appium/issues/12892
+    private static final Integer DEFAULT_VALUE = 70;
+    private Integer value = DEFAULT_VALUE;
+
+    public SnapshotMaxDepth() {
+        super(Integer.class, SETTING_NAME);
+    }
+
+    @Override
+    public Integer getValue() {
+        return value;
+    }
+
+    @Override
+    public Integer getDefaultValue() {
+        return DEFAULT_VALUE;
+    }
+
+    @Override
+    protected void apply(Integer value) {
```

Comment:
int value

Reply:
it would probably be a good idea to validate the value range here

Reply:
Thank you for your review!

> int value

I have a question, `AbstractSetting<T>` apply method is `apply(T value)`. Should I use `Integer value`? I encountered a build error `error: method does not override or implement a method from a supertype` with `int value`.
https://github.com/appium/appium-uiautomator2-server/blob/3725babe3f7e13cb6be3e9b8c9ffffc0ea000ccb/app/src/main/java/io/appium/uiautomator2/model/settings/AbstractSetting.java#L53

> it would probably be a good idea to validate the value range here

I'll add a validation.

Reply:
if the compiler complains then just leave Integer as is

Reply:
Add a validation. The range is 10 to 200. [8f855e2](https://github.com/appium/appium-uiautomator2-server/pull/517/commits/8f855e284e8b53f932b97b8249c308770e1562b3)